idx/metric_definition_test.go

```diff
@@ -0,0 +1,276 @@
+package idx
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"unsafe"
+
+	goi "github.com/robert-milan/go-object-interning"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+var defs []*MetricDefinition
+var tags []string
+var metName = "metric.names.can.be.a.bit.longer.than.normal.sometimes"
+var metNameRepeating = "metric.metric.metric.metric.metric.metric.metric.metric"
+var testSz string
+
+func genTags(num int) []string {
+	szs := make([]string, num)
+	for i := 0; i < len(szs); i++ {
+		szs[i] = fmt.Sprintf("key%d=val%d", i, i)
+	}
+	return szs
+}
+
+func genMetricDefinitionsWithSameName(num int, defs []*MetricDefinition) {
+	for i := 0; i < len(defs); i++ {
+		defs[i] = &MetricDefinition{
+			OrgId:      1,
+			Interval:   10,
+			LastUpdate: 0,
+			Partition:  1,
+		}
+		defs[i].SetTags(genTags(5))
+		defs[i].SetMType("rate")
+		defs[i].SetMetricName("anotheryetlonger.short.metric.name")
+		defs[i].SetUnit("test")
+		defs[i].SetId()
+	}
+}
+
+func genMetricDefinitionsWithoutTags(num int, defs []*MetricDefinition) {
+	for i := 0; i < len(defs); i++ {
+		defs[i] = &MetricDefinition{
+			OrgId:      1,
+			Interval:   10,
+			LastUpdate: 0,
+			Partition:  1,
+		}
+		defs[i].SetMType("rate")
+		defs[i].SetMetricName(fmt.Sprintf("metric%d", i))
+		defs[i].SetUnit("test")
+		defs[i].SetId()
+	}
+}
+
+func TestCreateDeleteMetricDefinition10(t *testing.T) {
+	testCreateDeleteMetricDefinition(t, 10)
+}
+
+func TestCreateDeleteMetricDefinition1000(t *testing.T) {
+	testCreateDeleteMetricDefinition(t, 1000)
+}
+
+func testCreateDeleteMetricDefinition(t *testing.T, num int) {
+	IdxIntern = goi.NewObjectIntern(nil)
+	defs = make([]*MetricDefinition, num)
+	name := "anotheryetlonger.short.metric.name"
+
+	genMetricDefinitionsWithSameName(num, defs)
+
+	originalNameAddress := defs[0].Name.Nodes()[0]
+
+	Convey("When creating MetricDefinitions", t, func() {
+		Convey(fmt.Sprintf("number of definitions should be %d", num), func() {
+			So(defs, ShouldHaveLength, num)
+		})
+		Convey(fmt.Sprintf("reference counts should be at %d", num), func() {
```

Comment:
all these other cases here seem to the test go-object-interning rather than anything index specific.
I think these tests should just be moved to that package (and be a bit more abstract, e.g. just strings, not MetricDefinitions)
then from the idx package perspective, there is very little idx specific (only the TestTagKeyValuesAndNameWithTags test remains I think?)

Reply:
I already test for all of these things in the interning library. I wanted to make sure there wasn't something else happening in the new methods I wrote for Metrictank which use the interning library methods.